spread.yaml

```diff
@@ -47,30 +52,41 @@ prepare: |
     apt install -y git
     go get ./tests/lib/snapbuild
 
-    # Snapshot the state including core.
-    if [ ! -f snapd-state.tar.gz ]; then
-        ! snap list | grep core || exit 1
-        snap install hello-world
-        snap list | grep core
-        snap remove hello-world
-        rmdir /snap/hello-world # Should be done by snapd.
-
-        systemctl stop snapd
-        systemctl daemon-reload
-        mounts="$(systemctl list-unit-files | grep '^snap[-.].*\.mount' | cut -f1 -d ' ')"
-        services="$(systemctl list-unit-files | grep '^snap[-.].*\.service' | cut -f1 -d ' ')"
-        for unit in $services $mounts; do
-            systemctl stop $unit
-        done
-        tar czf snapd-state.tar.gz /var/lib/snapd /snap /etc/systemd/system/snap-*core*.mount
-        systemctl daemon-reload # Workaround for http://paste.ubuntu.com/17735820/
-        for unit in $mounts $services; do
-            systemctl start $unit
-        done
-    fi
-
 suites:
-    tests/:
+    tests/main/:
         summary: Full-system tests for snapd
+        prepare: |
+            apt install -y ${SPREAD_PATH}/../snapd_*.deb
+            # Snapshot the state including core.
+            if [ ! -f $SPREAD_PATH/snapd-state.tar.gz ]; then
+                ! snap list | grep core || exit 1
+                snap install hello-world
+                snap list | grep core
+                snap remove hello-world
+                rmdir /snap/hello-world # Should be done by snapd.
+
+                systemctl stop snapd
+                systemctl daemon-reload
+                mounts="$(systemctl list-unit-files | grep '^snap[-.].*\.mount' | cut -f1 -d ' ')"
+                services="$(systemctl list-unit-files | grep '^snap[-.].*\.service' | cut -f1 -d ' ')"
+                for unit in $services $mounts; do
+                    systemctl stop $unit
+                done
+                tar czf $SPREAD_PATH/snapd-state.tar.gz /var/lib/snapd /snap /etc/systemd/system/snap-*core*.mount
+                systemctl daemon-reload # Workaround for http://paste.ubuntu.com/17735820/
+                for unit in $mounts $services; do
+                    systemctl start $unit
+                done
+            fi
+        restore: |
+            $SPREAD_PATH/tests/lib/reset.sh
+            apt purge -y snapd || true
         restore-each: |
             $SPREAD_PATH/tests/lib/reset.sh --reuse-core
+
+    tests/upgrade/:
+        summary: Tests for snapd upgrade
+        restore:
+            apt purge -y snapd || true
+        restore-each: |
+            $SPREAD_PATH/tests/lib/reset.sh
```

Comment:
Maybe use `$TESTSLIB` here? Also, are `SPREAD_PATH` and `PROJECT_PATH` different?